demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js

```diff
@@ -1,16 +1,106 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {Carousel, Image} from 'react-bootstrap';
 import {Link} from 'react-router-dom';
 import slideOne from '../../../images/slide_one.jpg';
 import slideTwo from '../../../images/slide_two.jpg';
 import slideThree from '../../../images/slide_three.jpg';
+import {event} from '../../../lib/gtag.js';
+import {CodeModal} from '../../../components/CodeModal/CodeModal.js';
+import {getItemParameters, getEventCodeSnippet, getMeasureCodeSnippet} from '../../../utils.js';
 import './HomeCarousel.css';
 
 /**
  * Creates slideshow component for showcasing our latest deals and offerings.
- * @return {!JSX} The component.
+ * @return {!JSX}
  */
 export function HomeCarousel() {
+  const viewPromotionParameters = {
+    items: [
+      {
+        ...getItemParameters('7ba94'),
+        promotion_name: 'Home Carousel',
+      },
+      {
+        ...getItemParameters('hjdf7'),
+        promotion_name: 'Home Carousel',
+        discount: 25,
+      },
+      {
+        ...getItemParameters('3h488'),
+        promotion_name: 'Home Carousel',
+      },
+    ],
+    promotion_name: 'Home Carousel',
+  };
+
+  // Send view_promotion event when the carousel loads for the first time.
+  useEffect(() => {
+    event('view_promotion', viewPromotionParameters);
+  }, []);
+
+  const carouselCodeModal =
+      <CodeModal
+        popupId="view_promotion"
+        gtagCode={
+          getEventCodeSnippet('view_promotion', viewPromotionParameters)
+        }
+        measureCode={getMeasureCodeSnippet()}
+      />;
+
+  /**
+   * Collects a single promotional item's standard parameters and
+   * returns it as an object with items attribute mapped to
+   * the items array containing a single promotional item.
+   * @param {string} itemId
+   * @param {number=} discount Price discount of item (if any)
+   * @return {{items: Array<!ItemParameters>}} An object containing
+   *     an items attibute mapped to an array of {@link ItemParameters}
+   */
+  function getSelectPromotionParameters(itemId, discount = 0) {
+    return {
+      items: [
+        {
+          ...getItemParameters(itemId),
+          promotion_name: 'Home Carousel',
+          // if discount is positive, include it in the item parameter
+          ...(discount > 0 && {discount}),
```

Comment:
Although it seems like a convenient shortcut, it requires an assumption that ...(false) behaves correctly on all browsers. 
So just to be extra cautious and don't require from a reader to go over edge-cases in the specification, I'd recommend using `discount > 0 ? {discount} : {}`

Reply:
This is something I hadn't considered before - thanks for the suggestion!